src/engine/controls/cuecontrol.cpp

```diff
@@ -467,21 +467,30 @@ void CueControl::loadCuesFromTrack() {
         return;
 
     for (const CuePointer& pCue: m_pLoadedTrack->getCuePoints()) {
-        if (pCue->getType() == mixxx::CueType::MainCue) {
+        switch (pCue->getType()) {
+        case mixxx::CueType::MainCue:
             DEBUG_ASSERT(!pLoadCue); // There should be only one MainCue cue
             pLoadCue = pCue;
-        } else if (pCue->getType() == mixxx::CueType::Intro) {
+            break;
+        case mixxx::CueType::Intro:
             DEBUG_ASSERT(!pIntroCue); // There should be only one Intro cue
             pIntroCue = pCue;
-        } else if (pCue->getType() == mixxx::CueType::Outro) {
+            break;
+        case mixxx::CueType::Outro:
             DEBUG_ASSERT(!pOutroCue); // There should be only one Outro cue
             pOutroCue = pCue;
-        } else if (pCue->getType() == mixxx::CueType::HotCue && pCue->getHotCue() != Cue::kNoHotCue) {
+            break;
+        case mixxx::CueType::HotCue:
+        case mixxx::CueType::Loop: {
```

Comment:
This hack makes me nervous. What happens when a Loop cue is saved after CueControl treats it as a HotCue? Is the end position lost?

Reply:
No, CueControl works on actual `Cue` objects, not some abstraction. The only functions that touch the hotcue position are `hotcueSet` and `hotcueClear`. It is possible to clear the saved loop and set a regular cue at its index, but that is intended.

For testing I modified label and color of a saved loop and the positions are unchanged (except rounding it up/down):
```sql
sqlite> SELECT * FROM cues WHERE track_id = (SELECT id FROM track_locations WHERE location = "/data/jan/Projects/serato-tags/shared/hotcues-and-loops.flac");
276|70|1|0|0|0|Start of Track|12592678
277|70|1|22402.8|0|1||16286234
278|70|1|1863136.8|0|3||16433939
279|70|1|11066630.4|0|5||13514142
280|70|1|18821086.2|0|7|End of Track|6824886
281|70|4|12539129.4|368146.800000001|8||16744448
282|70|4|8121456|736293.6|10|Electric Guitar|16744448
283|70|4|12539129|368147|11|Locked ILoop|16744448
sqlite> SELECT * FROM cues WHERE track_id = (SELECT id FROM track_locations WHERE location = "/data/jan/Projects/serato-tags/shared/hotcues-and-loops.flac");
276|70|1|0|0|0|Start of Track|12592678
277|70|1|22402.8|0|1||16286234
278|70|1|1863136.8|0|3||16433939
279|70|1|11066630.4|0|5||13514142
280|70|1|18821086.2|0|7|End of Track|6824886
281|70|4|12539129.4|368146.800000001|8||16744448
282|70|4|8121456|736294|10|Electric Guitar modified|11469004
283|70|4|12539129|368147|11|Locked ILoop|16744448
284|70|2|23174|0|-1||16744448
285|70|8|23174|18613774|-1||16744448
286|70|6|23174|0|-1||16744448
287|70|7|-1|18636948|-1||16744448
```

By the way, when you right-click the saved loop in the overview waveform and open it's cue menu, it even shows you the start/end position.

Reply:
And what happens to the end point when the user sets the cue to another position?

Reply:
How? I'm not aware it's possible to move cues. You can clear it and set a new one. This obviously deletes the saved loop.

Is there another way?